PR: https://github.com/odoo/odoo/pull/215789

From: edc6562db9a8eff0bf5b92bbda17d36af30927f3
From: alan-odoo
Date: 2025-09-06 21:40:47

Breaking data model changes scores: del:2 + add:0, change matches:
-    _inherit = 'res.users'
-    request_overtime = fields.Boolean(compute='_compute_request_overtime')

Total Changes: 33

[IMP] *: update the res.users form views

This commit modifies the fields on the preferences / profile form, which
allows users to read or modify their own data depending on their
permissions, and on the user form for the administrators. The aim of this
change is to display more relevant information to users.

The hr.res_users_view_form_simple_modif view has been removed and merged in
the mail_bot_hr.res_users_view_form_simple_modif view since it must not be
used anymore for the preferences form as this one must not have a different
display when the hr module is installed.

As some fields are removed from the forms and are not longer used, they are
also removed from the res.users model.

Enterprise PR: https://github.com/odoo/enterprise/pull/88518
Upgrade PR: https://github.com/odoo/upgrade/pull/8207

Task-4775990

Part-of: odoo/odoo#215789
Related: odoo/enterprise#88518
Related: odoo/upgrade#8207
Signed-off-by: Thibault Delavallee (tde) <tde@openerp.com>

================================= pseudo patch: =================================

--- a/addons/hr_holidays_attendance/models/__init__.py
+++ b/addons/hr_holidays_attendance/models/__init__.py
@@ -5,4 +5,3 @@ from . import hr_leave_accrual_plan_level
 from . import hr_leave_type
 from . import hr_leave
 from . import ir_ui_menu
-from . import res_users

--- a/addons/hr_holidays_attendance/models/res_users.py
+++ b/None
@@ -1,32 +0,0 @@
-# -*- coding: utf-8 -*-
-# Part of Odoo. See LICENSE file for full copyright and licensing details.
-
-from odoo import api, fields, models
-
-
-class ResUsers(models.Model):
-    _inherit = 'res.users'
-
-    request_overtime = fields.Boolean(compute='_compute_request_overtime')
-
-    @property
-    def SELF_READABLE_FIELDS(self):
-        return super().SELF_READABLE_FIELDS + ['request_overtime']
-
-    @api.depends_context('uid')
-    @api.depends('total_overtime')
-    def _compute_request_overtime(self):
-        is_holiday_user = self.env.user.has_group('hr_holidays.group_hr_holidays_user')
-        time_off_types = self.env['hr.leave.type'].search_count([
-            ('requires_allocation', '=', True),
-            ('employee_requests', '=', True),
-            ('overtime_deductible', '=', True)
-        ])
-        for user in self:
-            if user.total_overtime >= 1:
-                if is_holiday_user:
-                    user.request_overtime = True
-                else:
-                    user.request_overtime = time_off_types
-            else:
-                user.request_overtime = False
